src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8455,7 +8455,10 @@ private static CSharpSyntaxNode GetConditionalReceiverSyntax(ConditionalAccessEx
         private BoundExpression GetReceiverForConditionalBinding(ExpressionSyntax binding, DiagnosticBag diagnostics)
         {
             var conditionalAccessNode = SyntaxFactory.FindConditionalAccessNodeForBinding(binding);
-            Debug.Assert(conditionalAccessNode != null);
+            if (conditionalAccessNode == null)
+            {
+                return BadExpression(binding);
```

Comment:
Returning `BadExpression(binding)` felt to me like the most appropriate way to indicate an error without throwing here. But I don't have much experience with bound nodes, so: is this the right way to do it?

Reply:
This is perfectly fine, but I would add a comment explaining why this can happen: it can happen when someone attempts to speculate on an invocation inside a conditional access without speculating on the whole access itself.

---
In reply to: [465900509](https://github.com/dotnet/roslyn/pull/46565#discussion_r465900509) [](ancestors = 465900509)

Reply:
Done.